app/filters/SessionIdFilter.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import java.util.UUID
+
+import akka.stream.Materializer
+import com.google.inject.Inject
+import javax.inject.Singleton
+import play.api.mvc._
+import play.api.mvc.request.{Cell, RequestAttrKey}
+import uk.gov.hmrc.http.{SessionKeys, HeaderNames => HMRCHeaderNames}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
```

Comment:
I might need you to go over this file, I can gather what it's doing but it's not something I've seen added before. I'd just like to understand if it's needed or not.

Reply:
This is injected to create the session id instead we doing it manually on redirect in SessionIdentificationAction.